### PR TITLE
feat(endorsement-system): Data cleanup worker

### PR DESCRIPTION
**Pull request by rafnarnason**
this PR introduces a cleanup worker to fix data for endorsements
A bug released last may has populated to locality database field with wrong data
the cleanup worker can find these easily and fix them by requesting the right data and updating those rows

@coderabbitai ignore

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced a cleanup worker setup for the endorsement system to enhance operational efficiency.
  - Added a dedicated "cleanup" target to the project configuration for better maintenance.
  - Implemented the `EndorsementSystemCleanupWorkerModule` to modularize cleanup operations.
  - Created a cleanup service to update endorsement records by correcting invalid locality codes.
  - Added an asynchronous cleanup function to manage cleanup tasks within the endorsement system.

- **Bug Fixes**
  - Adjusted email configuration handling to prevent unintended behavior when environment variables are not set.

- **Documentation**
  - Updated initialization logic to support the new cleanup job type in the application setup process.

### CODE REVIEW

**Review by joningi98 — APPROVED** [review]

**Comment by rafnarnason** [comment]
@coderabbitai pause

**Comment by rafnarnason** [comment]
@coderabbitai resolve

File: apps/services/endorsements/api/infra/endorsement-system-api.ts [hunk 1]
```diff
@@ -45,3 +45,31 @@ export const serviceSetup =
...
+      NATIONAL_REGISTRY_B2C_CLIENT_SECRET:
+        '/k8s/api/NATIONAL_REGISTRY_B2C_CLIENT_SECRET',
+    })
+    .xroad(Base, Client, NationalRegistry, NationalRegistryB2C)
+    .liveness('/liveness')
+    .readiness('/liveness')
+    .extraAttributes({
+      dev: { schedule: '@hourly' },
+      staging: { schedule: '@hourly' },
+      prod: { schedule: '@hourly' },
```
Comment (AndesKrrrrrrrrrrr) [inline apps/services/endorsements/api/infra/endorsement-system-api.ts:74]: This is a bit aggressive for slow cleanups. Would e.g. `@daily` be sufficient?

Comment (AndesKrrrrrrrrrrr) [inline apps/services/endorsements/api/infra/endorsement-system-api.ts:74]:
```suggestion
      dev: { schedule: '@daily' },
      staging: { schedule: '@daily' },
      prod: { schedule: '@daily' },
```

File: apps/services/endorsements/api/infra/endorsement-system-api.ts [hunk 2]
```diff
@@ -45,3 +45,31 @@ export const serviceSetup =
...
+    .command('node')
+    .args('--no-experimental-fetch', 'main.js', '--job=cleanup')
+    .db({ name: 'services-endorsements-api' })
+    .migrations()
+    .secrets({
+      IDENTITY_SERVER_CLIENT_SECRET:
+        '/k8s/endorsement-system-api/IDS-shared-secret',
+      NATIONAL_REGISTRY_B2C_CLIENT_SECRET:
+        '/k8s/api/NATIONAL_REGISTRY_B2C_CLIENT_SECRET',
+    })
```
Comment (AndesKrrrrrrrrrrr) [inline apps/services/endorsements/api/infra/endorsement-system-api.ts:67]: Please double and triple check that these are all the secrets you'll need :pray:

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
`infra/` files LGTM, though I don't have domain knowledge on what access & secrets you'll need.

**Review by AndesKrrrrrrrrrrr — CHANGES_REQUESTED** [review]
You'll need to add your worker to the `infra/src/uber-charts/islandis.ts` file, see [here](https://github.com/island-is/island.is/pull/13530/files).

Thank you for pointing this out :heart:

File: apps/services/endorsements/api/infra/endorsement-system-api.ts (hunk 1 above)
Comment (rafnarnason) [reply apps/services/endorsements/api/infra/endorsement-system-api.ts:74]: no, this is a one off fixer, so we want to see bad rows found when it runs the first time, and see 0 rows found when it runs then next time after that

Comment (rafnarnason) [reply apps/services/endorsements/api/infra/endorsement-system-api.ts:74]: hourly is the plan

File: apps/services/endorsements/api/infra/endorsement-system-api.ts (hunk 2 above)
Comment (rafnarnason) [reply apps/services/endorsements/api/infra/endorsement-system-api.ts:67]: works locally with only these 2 in the environment

File: apps/services/endorsements/api/infra/endorsement-system-api.ts [hunk 3]
```diff
@@ -45,3 +45,31 @@ export const serviceSetup =
...
       .readiness('/liveness')
+
+const serviceName = 'endorsement-system-api'
+const serviceCleanupWorkerName = `${serviceName}-cleanup-worker`
+export const endorsementSystemCleanUpWorkerSetup = (): ServiceBuilder<
+  typeof serviceCleanupWorkerName
+> =>
+  service(serviceCleanupWorkerName)
+    .image('services-endorsements-api')
+    .namespace(serviceName)
```
Comment (AndesKrrrrrrrrrrr) [inline apps/services/endorsements/api/infra/endorsement-system-api.ts:56]:
```suggestion
    .namespace('endorsement-system')
```

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
LGTM

Just one comment that need fixing

File: apps/services/endorsements/api/infra/endorsement-system-api.ts (hunk 3 above)
Comment (rafnarnason) [reply apps/services/endorsements/api/infra/endorsement-system-api.ts:56]: well spotted

**Pull request merged**